--- vfmg-0.9.17/vfmg~	Mon Sep  6 11:59:25 2004
+++ vfmg-0.9.17/vfmg	Mon Sep  6 11:59:25 2004
@@ -712,7 +712,8 @@
 			$file=$name;
 			$file=~s/\s+/_/g;
 			$level.=".$file";
-			print "$name\"	Popup $level\n\n";
+			print "$name\"	Popup $level\n";
+			print "DestroyMenu recreate $level\n\n";
 			print "AddToMenu $level	\"$name\"	Title\n";
 			fvwm2($menu[$no]{$entry}[0]);
 			$level=~s/\.$file$//;
