--- vfmg-0.9.95/vfmg~	2011-02-22 12:03:59.971803362 +0100
+++ vfmg-0.9.95/vfmg	2011-02-22 12:16:50.471781710 +0100
@@ -477,8 +477,14 @@
 
 # read XDG menu specification
 my $file;
+my $menu_file;
+if ($ENV{'XDG_MENU_PREFIX'}) {
+    $menu_file = "$ENV{'XDG_MENU_PREFIX'}applications.menu"
+} else {
+    $menu_file = "applications.menu";
+}
 foreach my $tmp (@xdg_config_dirs) {
-	open F_IN, "$tmp/applications.menu" or next;
+	open F_IN, "$tmp/$menu_file" or next;
 	local $/ = undef;
 	($file = <F_IN>) =~ y/ \t\r\n//d;
 	$file =~ s/<!--.*?-->//g;
--- /home/users/baggins/devel/PLD/rpm/BUILD/vfmg-0.9.95/vfmg.sysconfig~	2005-12-07 17:22:27.000000000 +0100
+++ /home/users/baggins/devel/PLD/rpm/BUILD/vfmg-0.9.95/vfmg.sysconfig	2011-02-22 13:26:14.669831162 +0100
@@ -9,5 +9,7 @@
 # export VFMG_TERM="aterm -name xterm-%1 -e %2"
 # export LANG=pl_PL
 
+# export XDG_MENU_PREFIX="applnk-"
+
 # This must be last line !
 # vi:syntax=sh:tw=78:ts=8:sw=4
