--- vfmg-0.9.18/vfmg.orig	2004-10-14 22:44:54.821663680 +0200
+++ vfmg-0.9.18/vfmg	2004-10-14 23:02:52.508978912 +0200
@@ -867,7 +867,7 @@
 	open $F_OUT, ">> $dir/$_[1].menu" or warn "$dir/$_[1].menu: $!\n";
 	print $F_OUT "\"$_[2]\"\n";
 	foreach my $entry(sort keys %{$menu[$no]}) {
-		$name=$entry;
+		$name=encode($o_enc,$entry);
 		if($menu[$no]{$entry}[0]<0) {
 			$name=~s/\"/\\\"/g;
 			print $F_OUT "\"$name\" \"$menu[$no]{$entry}[1]\" exec \"$menu[$no]{$entry}[2]\"\n";
@@ -955,6 +955,10 @@
 eval "use open OUT => ':locale'" unless $o_enc;
 binmode(STDOUT, ":encoding($o_enc)") if $o_enc;
 
+require I18N::Langinfo;
+I18N::Langinfo->import(qw(langinfo CODESET));
+$o_enc=langinfo(CODESET()) unless $o_enc;
+
 if($o_output eq "icewm") {
 	icewm($o_strip);
 	exit;
