--- vfmg-0.9.18/vfmg.orig	2004-12-07 19:12:25.793620512 +0100
+++ vfmg-0.9.18/vfmg	2004-12-07 19:13:33.724293480 +0100
@@ -74,7 +74,7 @@
 @tmp=split(/:+/,$ENV{'XDG_DATA_DIRS'}) if $ENV{'XDG_DATA_DIRS'};
 my @xdg_data_dirs=grep -d, $tmp,@tmp;
 
-$tmp="$ENV{'HOME'}/.config";
+#$tmp="$ENV{'HOME'}/.config";
 $tmp="$ENV{'XDG_CONFIG_HOME'}" if $ENV{'XDG_CONFIG_HOME'};
 @tmp=("/etc/xdg");
 @tmp=split(/:+/,$ENV{'XDG_CONFIG_DIRS'}) if $ENV{'XDG_CONFIG_DIRS'};
