--- vfmg-0.9.18/vfmg.init~	2004-09-17 03:50:12.000000000 +0200
+++ vfmg-0.9.18/vfmg.init	2004-09-17 03:50:12.000000000 +0200
@@ -20,11 +20,13 @@
   start|restart|reload|force-reload)
 	if is_yes "$FLUXBOX" && [ -f /etc/X11/fluxbox/menu2 ]; then
 			show "Regenerating fluxbox menu"; busy
-			if vfmg -i -f -x -c -s fluxbox > /etc/X11/fluxbox/menu2 2>/dev/null; then ok; else fail; fi
+			if vfmg -i -f -x -c -s fluxbox > /etc/X11/fluxbox/menu2. 2>/dev/null \
+				&& mv /etc/X11/fluxbox/menu2. /etc/X11/fluxbox/menu2; then ok; else fail; fi
 	fi
 	if is_yes "$XFCE4" && [ -f /etc/xfce4/menu2.xml ]; then
 			show "Regenerating XFCE4 menu"; busy
-			if vfmg -i -f -x -c -u -m xfce4 > /etc/xfce4/menu2.xml 2>/dev/null; then ok; else fail; fi
+			if vfmg -i -f -x -c -u -m xfce4 > /etc/xfce4/menu2.xml. 2>/dev/null \
+				&& mv -f /etc/xfce4/menu2.xml. /etc/xfce4/menu2.xml; then ok; else fail; fi
 	fi
 	if is_yes "$FVWM2" && [ -f /etc/X11/fvwm2/fvwm2.menu2 ]; then
 			show "Regenerating FVWM2 menu"; busy
